File: flask/blog/blog/data/comment_data.py
from typing import Any
from data import database

def comment_by_id(comment_id: int) -> dict[str,Any]:
    comment= database.find_single(
        """SELECT idcomment,content,post_idpost,user_iduser
        FROM public.comment
        WHERE idcomment=%s;""",
        [str(comment_id)],
    )
    return {
        "idcomment": comment[0],
        "content":comment[1],
        "postIdpost":comment[2],
        "userIduser":comment[3]
    }


def comments_of_post(post_id: int) -> list[Any]:
    comments= database.find_all(
        """SELECT idcomment,content,post_idpost,user_iduser
        FROM public.comment WHERE post_idpost=%s;""",
        [str(post_id)],
    )

    reshaped= list(map(lambda comment: {
        "idcomment": comment[0],
        "content":comment[1],
        "postIdpost":comment[2],
        "userIduser":comment[3],
    },comments))

    return reshaped

def new_comment(comment_content: str,post_id: int, user_id: int) -> int:
    return database.insert_data_returning_id(
        """INSERT INTO public.comment (content,post_idpost,user_iduser)
        VALUES (%s,%s,%s)
        RETURNING idcomment;""",
        [comment_content, str(post_id), str(user_id)],
    )


def delete_comment(comment_id: int, user_id: int):
    database.delete_all(
        """DELETE FROM public.comment
        WHERE idcomment=%s
        AND user_iduser=%s;""",
        [str(comment_id), str(user_id)],
    )

File: spring/blog/src/main/java/development/blog/data/CommentData.java
package development.blog.data;

import development.blog.representations.request.NewComment;
import development.blog.representations.response.CommentResponse;
import org.springframework.stereotype.Service;

import java.sql.*;
import java.util.ArrayList;

@Service
public class CommentData {
    Connection connection;
    CommonData commonData;

    public CommentData(Database database, CommonData commonData) {
        this.connection = database.getConnection();
        this.commonData = commonData;
    }

    public ArrayList<CommentResponse> commentsOfPost(Long id) throws SQLException {
        final String sql = "SELECT idcomment,content,post_idpost,user_iduser FROM public.comment WHERE post_idpost=?;";
        PreparedStatement statement =
                this.connection.prepareStatement(sql);
        statement.setLong(1, id);
        ResultSet rs = statement.executeQuery();
        ArrayList<CommentResponse> comments = new ArrayList<CommentResponse>();

        while (rs.next()) {
            comments.add(new CommentResponse(
                    rs.getLong(1),
                    rs.getString(2),
                    rs.getLong(3),
                    rs.getLong(4)
            ));
        }
        return comments;
    }

    public CommentResponse addComment(NewComment newComment, Long userId) throws SQLException {
        final String sql = "INSERT INTO public.comment (content,post_idpost,user_iduser) VALUES (?,?,?);";
        PreparedStatement statement =
                this.connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS);
        statement.setString(1, newComment.getContent());
        statement.setLong(2, newComment.getPostId());
        statement.setLong(3, userId);
        statement.executeUpdate();

        ResultSet rs = statement.getGeneratedKeys();
        rs.next();

        Long generatedId = rs.getLong(1);
        return getCommentById(generatedId);
    }

    public CommentResponse getCommentById(Long id) throws SQLException {
        final String sql = "SELECT idcomment,content,post_idpost,user_iduser FROM public.comment WHERE idcomment=?;";
        PreparedStatement statement =
                this.connection.prepareStatement(sql);
        statement.setLong(1, id);
        ResultSet rs = statement.executeQuery();
        rs.next();

        return new CommentResponse(
                rs.getLong(1),
                rs.getString(2),
                rs.getLong(3),
                rs.getLong(4)
        );
    }

    public void deleteComment(Long commentId, Long userId) throws SQLException {
        final String sql = "DELETE FROM public.comment where idcomment=? AND user_iduser=?;";
        PreparedStatement statement =
                this.connection.prepareStatement(sql);
        statement.setLong(1, commentId);
        statement.setLong(2, userId);
        statement.executeUpdate();
    }
}
